pyproject.toml

```diff
@@ -19,39 +19,27 @@ classifiers = [
 ]
 dynamic = ["version"]
 dependencies = [
-    'asciitree',
-    'numpy>=1.24',
-    'fasteners',
-    'numcodecs[msgpack]>=0.10.0',
-    'fsspec>2024',
-    'crc32c',
-    'zstandard',
-    'typing_extensions',
-    'donfig',
-    'pytest',
-    'universal_pathlib>=0.2.0',
+    "numpy>=1.24",
     "zarr>=3.0.3,<3.1",
 ]
 
-[project.optional-dependencies]
+[dependency-groups]
 test = [
-    "coverage",
+    "aiohttp",
+    "fsspec>2024",
+    "numcodecs>=0.16.1",
     "pytest",
-    "pytest-cov",
-    "msgpack",
-    "lmdb",
-    "s3fs",
     "pytest-asyncio",
-    "moto[s3]",
-    "flask-cors",
-    "flask",
-    "requests",
-    "mypy",
-    "hypothesis",
     "pytest-xdist",
 ]
-dev = ["maturin", "pip", "pre-commit"]
 doc = ["sphinx>=7.4.6", "myst-parser"]
+dev = [
+    "maturin",
+    "pip",
+    "pre-commit",
+    {include-group = "test"},
+    {include-group = "doc"},
+]
```

Comment:
@flying-sheep has a good reason to keep `test` and `doc` out of `dev` but i can't remember it off hand.

Now that I'm looking at this, I'm also curious why we have `pre-commit` as dependency here but not in our other projects @flying-sheep ? I never really thought twice about it (I believe I have it globally installed).

Also why `pip`?  I assume these both came from the maturin template, but still always good to get a @flying-sheep take